### PR TITLE
Fixed #5663 - AOR_Reports Date issues

### LINKED ISSUE #5663: AOR_Reports Date issues
#### Issue
There are still some serious issues with the date in the AOR_Reports module.
Using two dates in Conditions does break the whole function. The second date will not have any hour and minutes fields, and the fields of the first date will be reseted. The datepicker of the second condition additionaly does not work. When deleting the second date field, its not possible to save because of the error message 'Missing required field: Hours'. When only one date is used and it is set as a parameter, the date will be set to a wrong value if date format d.m.Y is used.
Picking date 01.04.2018 and after clicking "update" the date will be 04.01.2018.
Using the date format Y-m-d the does not resolve the issue, but opens another one. The date will be deleted when clicking update on the DetailView of the Reports module.

#### Expected Behavior
Using dates should work in the Reports module.

#### Actual Behavior
When using dates everythings seems to be messed up. 

#### Steps to Reproduce

1. Go to report module
2. Create condition
3. Add a date field and choose a date
4. Add another date field
5. After that, saving is not possible (If only one date field is used it can be saved)
Also, setting one date to parameters does work, but changing the date in DetailView does not take the right date.

#### Context

Unable to use date fields in the Report module as parameters. Unable to use more than one date as conditions.

#### Your Environment

SuiteCRM Version used: 7.10.3 (demo)
Browser name and version: Chrome 65.0.3325.181
Environment: demo.suiteondemand.com

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

Fixes issues with #5799 pr.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

File: include/utils.php [hunk 1]
```diff
@@ -1768,8 +1768,7 @@ function sugar_die($error_message, $exit_code = 1)
 {
     global $focus;
     sugar_cleanup();
-    //echo $error_message;
-    //die($exit_code);
+    echo $error_message;
```
Comment (samus-aran) [inline include/utils.php]: Why was this removed?

File: modules/AOR_Reports/AOR_Report.js [hunk 2]
```diff
@@ -52,51 +52,77 @@ $(document).ready(function () {
...
-    if($('#aor_conditions_value\\['+index+'\\]').hasClass('date_input')) { // only change to DB format if its a date
-      if ($('#aor_conditions_value\\[' + ln + '\\]').hasClass('date_input')) {
-        fieldInput = $.datepicker.formatDate('yy-mm-dd', new Date(fieldInput));
-      }
-    }
-    _form.append('<input type="hidden" name="parameter_value[]" value="'+fieldInput+'">');
+    appendHiddenFields(_form, ln, id);
   });
-
+    debugger;
```
Comment (samus-aran) [inline modules/AOR_Reports/AOR_Report.js]: Debugger?

File: modules/AOS_Products/vardefs.php [hunk 3]
```diff
@@ -40,6 +40,7 @@
 $dictionary['AOS_Products'] = array(
     'table' => 'aos_products',
     'audited' => true,
+    'duplicate_merge' => true,
```
Comment (samus-aran) [inline modules/AOS_Products/vardefs.php]: This not relevant? I'm assuming not your changes?

File: modules/AOW_WorkFlow/aow_utils.php [hunk 4]
```diff
@@ -305,14 +305,53 @@ function getValidFieldsTypes($module, $field){
...
+            . $module
+            . $view
+            . $alt_type
+            . $fieldname
+            . '.tpl';
+    }
+
     // set the filename for this control
-    $file = create_cache_directory('modules/AOW_WorkFlow/') . $module . $view . $alt_type . $fieldname . '.tpl';
+    $file = create_cache_directory('modules/AOW_WorkFlow/') . $module . $view . $alt_type . $fieldname . $aor_row . '.tpl';
```
Comment (samus-aran) [inline modules/AOW_WorkFlow/aow_utils.php]: A new tpl for each row?

File: modules/AOW_WorkFlow/aow_utils.php [hunk 5]
```diff
@@ -617,7 +656,9 @@ function getModuleField($module, $fieldname, $aow_field, $view='EditView',$value
  */
 function createBracketVariableAlias($variable)
 {
-    return str_replace('[', 'SCRMLSQBR', str_replace(']', 'SCRMRSQBR', $variable));
+    $replaceRightBracket = str_replace(']', '', $variable);
```
Comment (samus-aran) [inline modules/AOW_WorkFlow/aow_utils.php]: Anywhere else we know that uses this SCRMRSQBR

File: modules/Campaigns/Subscriptions.php [hunk 6]
```diff
@@ -157,37 +157,34 @@ function constructDDSubscriptionList($focus,$classname=''){
 }
 
 
-
-/*
+ /*
```
Comment (samus-aran) [inline modules/Campaigns/Subscriptions.php]: Not relevant?

File: include/Sugar_Smarty.php [hunk 7]
```diff
@@ -151,10 +151,10 @@ public function fetch($resource_name, $cache_id = null, $compile_id = null, $dis
         $state = new SuiteCRM\StateSaver();
         $state->pushErrorLevel('sugar_smarty_errors');
         
-        if (!(isset($sugar_config['developerMode']) && $sugar_config['developerMode'])) {
-            $level = isset($sugar_config['smarty_error_level']) ? $sugar_config['smarty_error_level'] : 0;
-            error_reporting($level);
-        }
+
```
Comment (samus-aran) [inline include/Sugar_Smarty.php]: Not relevant?

File: modules/AOW_WorkFlow/aow_utils.php (hunk 4 above)
Comment (daniel-samson) [reply modules/AOW_WorkFlow/aow_utils.php]: Yes, this ensures that the id and name are correct.

File: modules/AOW_WorkFlow/aow_utils.php (hunk 5 above)
Comment (daniel-samson) [reply modules/AOW_WorkFlow/aow_utils.php]: Only reports use this functionality

File: include/utils.php (hunk 1 above)
Comment (daniel-samson) [reply include/utils.php]: I didn't remove this. It must have come from an a previous commit message

File: include/Sugar_Smarty.php [hunk 8]
```diff
@@ -151,10 +151,10 @@ public function fetch($resource_name, $cache_id = null, $compile_id = null, $dis
...
+        $level = isset($sugar_config['smarty_error_level']) ? $sugar_config['smarty_error_level'] : 0;
+        error_reporting($level);
+
         $fetch = parent::fetch(get_custom_file_if_exists($resource_name), $cache_id, $compile_id, $display);
```
Comment (gymad) [inline include/Sugar_Smarty.php]: restore the error reporting level, after fetch

File: include/Sugar_Smarty.php (hunk 7 above)
Comment (daniel-samson) [reply include/Sugar_Smarty.php]: This  decouples the error messaging from debug mode. When you need to see the cached tpl, development mode will clear them. So this helps to see the errors.

**Comment by daniel-samson** [comment]
I am going to cherry pick my changes in a new pull request
